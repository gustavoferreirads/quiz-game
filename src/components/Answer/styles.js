import { StyleSheet } from 'react-native';

import { colors, fonts, metrics } from '../../styles';

const borderWidth = 2;
const styles = StyleSheet.create({
  container: {
    width: '100%',
    borderRadius: 40,
    borderStyle: 'solid',
    borderWidth,
    borderColor: colors.lighter,
    alignItems: 'center',
    flexDirection: 'row',
    padding: metrics.size,
    marginTop: 20,
  },
  contentOption: {
    borderStyle: 'solid',
    borderRadius: 32,
    width: 32,
    height: 32,
    borderWidth,
    borderColor: colors.primary,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 10,
  },
  option: {
    fontSize: fonts.big,
    color: colors.primary,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  text: {
    fontSize: fonts.regular,
    fontWeight: 'bold',
    color: colors.regular,
    textTransform: 'uppercase',
  },
});

export default styles;
