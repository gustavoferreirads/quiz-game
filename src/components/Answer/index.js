import React from 'react';
import * as PropTypes from 'prop-types';
import { Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';

const Answer = ({ option, text, onPress }) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    <View style={styles.contentOption}>
      <Text style={styles.option}>{option}</Text>
    </View>
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
);

Answer.defaultProps = {};
Answer.propTypes = {
  option: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default Answer;
