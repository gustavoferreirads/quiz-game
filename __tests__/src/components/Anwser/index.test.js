import React from 'react';
import { shallow } from 'enzyme/build';
import Answer from '../../../../src/components/Answer';

it('renders a Answer ', () => {
  expect(shallow(<Answer text="YES" option="A" onPress={() => jest.fn} />)).toMatchSnapshot();
});
