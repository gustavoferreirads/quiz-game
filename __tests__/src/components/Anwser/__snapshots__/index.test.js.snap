// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Answer  1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#eee",
      "borderRadius": 40,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "marginTop": 20,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#FF9982",
        "borderRadius": 32,
        "borderStyle": "solid",
        "borderWidth": 2,
        "height": 32,
        "justifyContent": "center",
        "marginRight": 10,
        "width": 32,
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#FF9982",
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      A
    </Component>
  </Component>
  <Component
    style={
      Object {
        "color": "#666",
        "fontSize": 14,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
  >
    YES
  </Component>
</TouchableOpacity>
`;
